src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CompilationContext.cs

```diff
@@ -1270,6 +1271,11 @@ private static DkmClrCompilationResultFlags GetLocalResultFlags(LocalSymbol loca
                     var instance = new DisplayClassInstanceFromParameter(parameter);
                     displayClassInstances.Add(new DisplayClassInstanceAndFields(instance));
                 }
+                else
+                {
+                    // Collect parameter names that will hide display class variables from outer scopes
+                    userDefinedParameterNames.Add(parameter.Name);
```

Comment:
>userDefinedParameterNames.Add(parameter.Name); [](start = 20, length = 46)

Locals also can cause similar shadowing, how are we handling them?

Reply:
Good catch. That scenario is busted as well.

---
In reply to: [574725130](https://github.com/dotnet/roslyn/pull/51129#discussion_r574725130) [](ancestors = 574725130)